File: fastapi-server/src/modules/health/application/health_check.py
"""ヘルスチェックユースケース（health モジュール内）。"""

from src.modules.health.domain import HealthStatus


class HealthCheckUseCase:
    """ヘルスチェックユースケース。

    依存関係を必要としないシンプルなユースケースです。
    """

    async def execute(self) -> HealthStatus:
        """ヘルスチェックを実行してステータスを返す。"""
        return HealthStatus.healthy()

File: go-server/internal/modules/health/application/health_check.go
package application

import "cariban/go-server/internal/modules/health/domain"

type HealthCheckUseCase struct{}

func NewHealthCheckUseCase() *HealthCheckUseCase {
	return &HealthCheckUseCase{}
}

func (u *HealthCheckUseCase) Execute() domain.HealthStatus {
	return domain.Healthy()
}
